src/analytics/analytics-step-tracker.ts

```diff
@@ -0,0 +1,450 @@
+import { LineItemMap } from '../cart';
+import { Checkout, CheckoutService } from '../checkout';
+import { ShopperCurrency, StoreProfile } from '../config';
+import { Coupon } from '../coupon';
+import { Order } from '../order';
+import { ShippingOption } from '../shipping';
+
+import { AnalyticsTracker } from './analytics-tracker-window';
+import StepTracker from './step-tracker';
+
+export interface StepTrackerConfig {
+    checkoutSteps?: AnalyticStepType[];
+}
+
+export type AnalyticStepType = 'customer' | 'shipping' | 'billing' | 'payment';
+
+const ORDER_ITEMS_STORAGE_KEY = 'ORDER_ITEMS';
+
+export enum AnalyticStepId {
+    CUSTOMER = 1,
+    SHIPPING,
+    BILLING,
+    PAYMENT,
+}
+
+const ANALYTIC_STEPS: { [key: string]: AnalyticStepId } = {
+    customer: AnalyticStepId.CUSTOMER,
+    shipping: AnalyticStepId.SHIPPING,
+    billing: AnalyticStepId.BILLING,
+    payment: AnalyticStepId.PAYMENT,
+};
+
+export default class AnalyticsStepTracker implements StepTracker {
+    private checkoutStarted: boolean = false;
+    private completedSteps: { [key: string]: boolean } = {};
+    private viewedSteps: { [key in AnalyticStepId]?: boolean; } = {};
+    private analyticStepOrder: AnalyticStepType[] = [
+        'customer',
+        'shipping',
+        'billing',
+        'payment',
+    ];
+
+    constructor(
+        private checkoutService: CheckoutService,
+        private storage: StorageFallback,
+        private analytics: AnalyticsTracker,
+        { checkoutSteps }: StepTrackerConfig = {}
+    ) {
+        if (checkoutSteps !== undefined) {
+            this.analyticStepOrder = checkoutSteps;
```

Comment:
We might need a tighter check - to ensure that the consumer can only pass in the allowed steps. This is because right now it only has a compiler check, which is not sufficient for apps that are not using TS.